drivers/Watchdog.h

```diff
@@ -100,6 +115,47 @@ class Watchdog {
      *         platform in milliseconds
      */
     static uint32_t max_timeout();
+protected :
+    /** Start an independent watchdog timer with specified parameters
+     *
+     *  @param timeout      Timeout of the watchdog in milliseconds
+     *
+     *  @return status WATCHDOG_STATUS_OK if the watchdog timer was started
+     *                 successfully. WATCHDOG_INVALID_ARGUMENT if one of the input
+     *                 parameters is out of range for the current platform.
+     *                 WATCHDOG_NOT_SUPPORTED if one of the enabled input
+     *                 parameters is not supported by the current platform.
+     */
+    watchdog_status_t start(const uint32_t timeout);
+
+    /** Stops the watchdog timer
+     *
+     * Calling this function will attempt to disable any currently running
+     * watchdog timers if supported by the current platform.
+     *
+     * @return Returns WATCHDOG_STATUS_OK if the watchdog timer was succesfully
+     *         stopped, or if the timer was never started. Returns
+     *         WATCHDOG_STATUS_NOT_SUPPORTED if the watchdog cannot be disabled
+     *         on the current platform.
+     */
+    mbed_error_status_t stop();
+
+private:
+    static rtos::Mutex _ThreadSafeLockMutex;//Thread safe mutex for concurrent access protection
+
+    typedef struct
+    {
+       osThreadId_t tid;
+       uint32_t bit_idx;
```

Comment:
Do you need this? Whatever position in the array can give the position in the bitmask?